src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/Huffman.cs

```diff
@@ -307,128 +400,147 @@ public static (uint encoded, int bitLength) Encode(int data)
         /// <returns>The number of decoded symbols.</returns>
         public static int Decode(ReadOnlySpan<byte> src, ref byte[] dstArray)
         {
+            // The code below implements the decoding logic for a HPack huffman encoded literal values.
+            // https://httpwg.org/specs/rfc7541.html#string.literal.representation
+            //
+            // To decode a symbol, we traverse the decoding lookup table tree by 8 bits for each lookup
+            // until we found a leaf - which contains decoded symbol (octet)
+            //
+            // see comments in GenerateDecodingLookupTree() describing decoding table
+
             Span<byte> dst = dstArray;
             Debug.Assert(dst != null && dst.Length > 0);
 
+            if (s_decodingTree == null)
+            {
+                Volatile.Write(ref s_decodingTree, GenerateDecodingLookupTree());
+            }
```

Comment:
Consider storing s_decodingTree into a local and then using that local instead of the static field on all future accesses in this method.

Reply:
done